api/src/transactions.rs

```diff
@@ -1177,7 +1177,7 @@ impl TransactionsApi {
         txn: SignedTransaction,
     ) -> SimulateTransactionResult<Vec<UserTransaction>> {
         // Transactions shouldn't have a valid signature or this could be used to attack
-        if txn.signature_is_valid() {
+        if txn.verify_signature().is_ok() {
```

Comment:
@gregnazario do we have a test for the simulation API?

Reply:
@vgao1996 I added a pair of tests in this PR, but otherwise I haven't seen simulation being tested anywhere